### PR TITLE
CI matrix update

### LINKED ISSUE #154: Enable and fix integration tests against devel
##### SUMMARY
The integration tests against `devel` were disabled in https://github.com/ansible-collections/community.mysql/pull/153  presumably because of https://github.com/ansible-collections/overview/issues/45#issuecomment-825453639

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Fixes https://github.com/ansible-collections/community.mysql/issues/154
Fixes https://github.com/ansible-collections/community.mysql/pull/162

CI matrix update
- Add testing against `stable-2.12`
- Add testing against `devel`
- Add `Python 3.8`
- Fix related failings

### CODE REVIEW

**Comment by Andersson007** [comment]
@Jorge-Rodriguez @unnecessary-username (thank you for exclusion statements in Jorge's PR) could you please take a look?

They are green but I'm not 100% sure I kept the initial logic.

**Comment by Jorge-Rodriguez** [comment]
I'll have a look, hold on for my feedback if you're not in too much of a hurry

**Comment by Andersson007** [comment]
> I'll have a look, hold on for my feedback if you're not in too much of a hurry

sure, will be waiting then:)

**Review by unnecessary-username — APPROVED** [review]
As far as I could come up with, I think it works as expected.
Thanks for the contribution.

File: tests/integration/targets/setup_mysql/tasks/install.yml [hunk 1]
```diff
@@ -28,6 +28,27 @@
 - name: "{{ role_name }} | install | install python packages"
   pip:
     name: "{{ python_packages }}"
+  register: connector
+
+- name: Debug connector var content
```
Comment (Jorge-Rodriguez) [inline tests/integration/targets/setup_mysql/tasks/install.yml:33]: These tasks seem superfluous as the value is presented on the output of the previous task.

**Review by Jorge-Rodriguez — APPROVED** [review]
I love the new conditionals. Thank you @unnecessary-username @Andersson007

File: tests/integration/targets/setup_mysql/tasks/install.yml (hunk 1 above)
Comment (Andersson007) [reply tests/integration/targets/setup_mysql/tasks/install.yml:33]: @Jorge-Rodriguez thanks for spotting! I'll remove it in a separate PR without reviewing - wouldn't like to wait again for all those checks get passed:)

**Comment by Andersson007** [comment]
@Jorge-Rodriguez @unnecessary-username thanks for reviewing!

File: tests/integration/targets/setup_mysql/tasks/install.yml (hunk 1 above)
Comment (Andersson007) [reply tests/integration/targets/setup_mysql/tasks/install.yml:33]: heh, anyway we have to wait for them there:)))

**Pull request merged**